0000-product.md

```diff
@@ -0,0 +1,504 @@
+otherwise the transaction is invalid.
+
+The inputs for ProductDeleteAction must include:
+
+* Address of the Agent submitting the transaction 
+* Address of the Organization the Product is being deleted for Address of the Product to be deleted
+
+The outputs for ProductUpdateAction must include:
+
+* Address of the Product to be deleted
+
+### Defined GS1 Properties
+
+An initial set of GS1 properties will be predefined within Grid. Each property
+will have a property definition of the following format:
+
+``` 
+PropertyDefinition( 
+    name="<GS1 Property>",
+    data_type=PropertyDefinition.DataType.STRING, 
+    required=False, 
+    description="A description of the GS1 data.") 
+```
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+
+Each GTIN is derived from three components. A GS1 Global Company Prefix (which
+also identifies the owner of the GTIN), and item reference number, and a check
+digit. This RFC duplicates the organization identification with a Grid-specific
+owner field to tie the product to organizations managed in Pike. In the future,
+it may be good to use the GTIN’s organizational identifier directly.
+
+Currently, Pike does not provide a place to store a GS1 Company Prefix, and the
+permissions around who is authorized to provide this prefix is complicated and
+not a direct extension of how Pike currently works. As such it is out of the
+scope of this RFC.
+
+Instead, Pike’s organization definition will be extended to include metadata,
+similar to Agent’s metadata. Metadata is a Key,Value store of arbitrary data.
+The GS1 Company Prefix will be stored in the metadata under
+“gs1_company_prefix”.
+
+Solving how to properly provide GS1 Company Prefixes to a Pike Organization will
+be solved in a future RFC. This future RFC should integrate the concepts of GLN
+and the smart contract associated with grid gs1 product should be updated to
+reflect that integration.  `
+
+Trade items can include non-material goods, such as services, which will also
+need to be represented within Grid. This is not covered by this RFC.
+
+Some examples of non-material goods: 
+
+- Batch/Lot (LGTIN) or Serialized (SGTIN) 
+- Logistical Cases/Pallets - Higher levels of packaging of the items - some of
+  which are the trade unit between to parties although not necessarily the
+consumer unit. Identified sometimes with a GTIN but often times, due to their
+heterogeneous nature, with another GS1 identifier (SSCC).
+- Assets (returnable or fixed) - Identified with GRAI or GIAI - I presume like
+  services (GSRN) OUT
+
+To expand the product schema to support all GS1 properties as well as keeping it
+all organized, there will need to be some refactoring done at the grid primitive
+level to support lists in the schema.
+
+This implementation does not account for transfer-of-ownership scenarios, and
+should be implemented as a supplemental RFC at a later time. See:
+https://github.com/hyperledger/grid-rfcs/pull/5#discussion_r312211331
```

Comment:
This kind of breaks my mind. I don't think we should include links to PR comments from the doc being commented on (or at all really). I think dropping "See: " and then the link is sufficient.

Reply:
✅